fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/service/ShardSystem.java

```diff
@@ -0,0 +1,39 @@
+package ch.bbw.fabbwled.lands.service;
+
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class ShardSystem {
```

Comment:
good idea to enforce the game logic (no negative shards).
an alternative location could have been `PlayerSession.update()`.
if this class would be immutable (like a record and shard updates use methods like `ShardSystem addShards(...)`) it would be possible to keep a history of PlayerDto. This is required for example for blessings. :angel: